### PR TITLE
fix(proposals): harden packet lifecycle routing

### PR DESCRIPTION
**Pull request by jamesryancooper**
## What

Hardens the proposal packet lifecycle extension routing and instruction surfaces so agents fail closed when required packet, program, or finding inputs are missing, while preserving the shortened `/octon-proposal-packet-*` command surface.

Also hardens the shared GitHub/PR closeout lane so `autonomy:no-automerge` PRs remain manual-lane and the protected auto-merge boundary checks out the same-repo PR head before using branch-local runtime state.

## Why

The closeout review found two lifecycle-extension reliability gaps: route resolution could select operational bundles without required inputs, and validation did not cover the full command/skill/prompt route matrix.

Closeout then surfaced a shared PR automation failure: the auto-merge helper attempted a manual-lane PR and invoked the protected runtime boundary from the base `main` checkout, which can see stale generated runtime state instead of the PR branch state.

No-Issue: follow-up hardening from proposal packet lifecycle extension closeout.

## How

- Adds fail-closed routing cases for missing packet paths, program packet paths, and correction finding IDs.
- Keeps packet-path-only routing read-only by default through `explain-proposal-packet`.
- Preserves source-driven packet creation when a target packet path is provided alongside `source_kind`, and escalates finding-only inputs instead of falling through to an opaque no-route error.
- Extends command and skill metadata with program packet and finding input hints.
- Expands route-resolution validation across lifecycle actions, bundle aliases, proposal-program flows, aliases, and missing-input escalation.
- Refreshes generated extension, capability, pack-route, runtime-route, support-envelope, run-health, host projection, and publication receipt surfaces.
- Teaches the PR autonomy evaluator to honor the checked `autonomy:no-automerge` template request as a manual-lane gate.
- Makes the protected PR auto-merge workflow fetch the exact eligible same-repo PR head SHA before invoking the runtime-protected merge boundary.

## Profile Selection Receipt

- Semantic version source(s): repository release state at `octon-v0.6.50`.
- Release state (`pre-1.0` or `stable`): `pre-1.0`.
- `change_profile` (`atomic` or `transitional`): `atomic`.
- Hard-gate facts (downtime, coordination, migration/backfill, rollback, blast radius, compliance): no downtime, no migration/backfill, no runtime service coordination; additive lifecycle extension hardening plus shared PR closeout remediation.
- Tie-break status: no tie-break escalation required.
- `transitional_exception_note` (required when `pre-1.0` + `transitional`): n/a.

## Implementation Plan

- Workstreams and scope: harden lifecycle extension routing, update operator-facing command/skill metadata, refresh generated publication surfaces, add route-matrix validation, and correct shared PR closeout automation behavior exposed during closeout.
- Public interfaces/types/contracts affected: proposal packet lifecycle extension routing contract, command manifest fragment, skill registry fragment, generated effective extension/capability/runtime route state, generated support-envelope and run-health read models, host skill projections, PR auto-merge workflow, and PR autonomy policy evaluator output.
- Test scenarios: route-resolution matrix, pack-shape validation, authority-boundary validation, extension publication state, capability publication state, runtime route bundle state, runtime effective state, support-envelope reconciliation, run-health read models, architecture conformance, mission autonomy scenario, host projections, protected-CI wrapper validation, Git/GitHub workflow alignment, and PR autonomy policy manual-lane evaluation.
- Assumptions/defaults: packet-path-only requests remain read-only; program-packet-only requests escalate as ambiguous; GitHub/PR/worktree closeout remains delegated to the shared Octon contract.

## Impact Map (code, tests, docs, contracts)

- code/contracts: `.octon/inputs/additive/extensions/octon-proposal-packet-lifecycle/context/routing.contract.yml`, command/skill metadata, generated effective extension, capability, pack-route, runtime-route, support-envelope, and run-health state; `.github/workflows/pr-auto-merge.yml`; `.octon/framework/assurance/governance/_ops/scripts/evaluate-pr-autonomy-policy.sh`.
- tests: lifecycle route-resolution, pack-shape, authority-boundary validation surfaces, protected-CI wrapper test, and Git/GitHub workflow alignment validator.
- docs: extension-local routing and scenario guidance.
- generated/evidence: host projections plus extension, capability, pack-route, runtime-route, support-envelope, run-health, compatibility, and prompt-alignment publication receipts.

## Compliance Receipt

- [x] Selected exactly one execution profile before planning/implementation.
- [x] Applied release-maturity gate from semantic version.
- [x] Enforced pre-1.0 default (`atomic`) unless hard gates required `transitional`.
- [x] Included `transitional_exception_note` when required.
- [x] Included tie-break escalation when applicable.
- [x] Updated impacted contracts/docs/tests.
- [x] Honored charter change-control constraint (no direct principles charter edits without override evidence).

## Exceptions/Escalations

none.

## Tradeoffs

The lifecycle change keeps routing conservative instead of inferring missing operational inputs. That may require an agent to ask for the missing packet, program, or finding identifier, but it prevents accidental mutation or closeout routing with incomplete context.

The shared closeout remediation keeps GitHub/PR/worktree control in the existing shared automation contract instead of duplicating closeout logic inside the proposal packet lifecycle extension.

## Testing

- `bash .octon/inputs/additive/extensions/octon-proposal-packet-lifecycle/validation/tests/test-route-resolution.sh`
  - Passed: 179 Failed: 0
- `bash .octon/inputs/additive/extensions/octon-proposal-packet-lifecycle/validation/tests/test-pack-shape.sh`
  - Passed: 138 Failed: 0
- `bash .octon/inputs/additive/extensions/octon-proposal-packet-lifecycle/validation/tests/test-authority-boundaries.sh`
  - Passed: 4 Failed: 0
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-extension-publication-state.sh`
  - Validation summary: errors=0
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-host-projections.sh`
  - Validation summary: errors=0
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-capability-publication-state.sh`
  - Validation summary: errors=0
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-runtime-effective-route-bundle.sh`
  - Validation summary: errors=0
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-runtime-effective-state.sh`
  - Validation summary: errors=0
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-support-envelope-reconciliation.sh`
  - Validation summary: errors=0
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-run-health-read-model.sh`
  - Validation summary: errors=0
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-architecture-conformance.sh`
  - Validation summary: errors=0
- `bash .octon/framework/assurance/runtime/_ops/scripts/test-mission-autonomy-scenarios.sh`
  - Passed locally, including the workflow-bundle contract scenario.
- `bash .octon/framework/assurance/runtime/_ops/tests/test-protected-ci-live-wrapper.sh`
  - Passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-git-github-workflow-alignment.sh`
  - Validation summary: errors=0 warnings=0
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-execution-governance.sh`
  - Validation summary: errors=0
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-phase6-simplification-deletion.sh`
  - Validation summary: errors=0
- `bash .octon/framework/assurance/runtime/_ops/tests/test-git-github-workflow-alignment.sh`
  - Passed: 4 Failed: 0
- `bash .octon/framework/assurance/runtime/_ops/tests/test-material-side-effect-coverage-fixtures.sh`
  - Passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-authorization-boundary-coverage.sh`
  - Validation summary: errors=0
- PR #378 policy probe via `evaluate-pr-autonomy-policy.sh`
  - `status=staged`, `reason_code=PR_AUTONOMY_MANUAL_LANE_REQUESTED`, `manual_lane_requested=true`
- `git diff --check`
  - Passed.

## Rollout

n/a. This is extension metadata, routing, prompt, validation, generated-publication, evidence, and shared PR closeout automation hardening with no runtime service rollout.

## Convivial Purpose Check

- [x] Feature expands genuine user capability (not synthetic engagement).
- [x] Attention and interruption behavior are justified and user-controllable.
- [x] No manipulative patterns or dark-pattern mechanics are introduced.
- [x] Data collection/extraction risk is minimal and explicitly justified.

## Risk Rubric

- Risk class: [ ] Trivial [ ] Low [x] Medium [ ] High
- Rollback plan: revert the PR commits to restore the prior lifecycle extension routing behavior, generated publication/read-model state, and PR closeout automation behavior.
- Rollback handle: commits `cd0f7e289`, `a30a3379`, `9c916875`, `0034ab8`, `22955ef`, and `bf0e605`.
- Flags changed (name, owner, expiry, rollout): none.
- Autonomy eligibility: [ ] autonomy:auto-merge [x] autonomy:no-automerge

## Contracts and Threat Model

- OpenAPI/JSON-Schema changes: none.
- Threat model update/link: routing now fails closed for missing operational inputs; PR auto-merge now honors manual-lane requests and reads protected runtime state from the same-repo PR head before attempting merge; no durable authority, runtime policy, or PR/worktree control logic is introduced inside the extension.

## Observability and Performance

- Traces/logs/metrics for changed flows: PR auto-merge emits existing workflow logs and protected runtime artifacts; lifecycle extension changes have no runtime service telemetry path.
- Representative traces for high risk changes: n/a; medium-risk shared closeout automation hardening validated locally and through PR checks.
- Performance or bundle impact: n/a.

## License and Provenance

- New dependencies and licenses: none.
- Generated code/templates provenance notes: generated/effective extension state, capability state, pack routes, runtime route bundle, support-envelope reconciliation, run-health read models, host projections, and publication receipts were refreshed from repository-local publication scripts and validators.

## Checklist

- [x] Requirements met; edge cases handled
- [x] Security reviewed (authz, input validation, secrets)
- [x] Tests added or updated
- [x] Observability updated (logs, metrics, traces) if needed
- [x] No speculative abstractions or unnecessary complexity
- [x] Conventions followed; no drift introduced
- [x] Non-obvious decisions documented (comments, ADR)
- [x] Reviewer feedback addressed with fix, commit, push, reply; reviewer-owned threads left for reviewer or maintainer confirmation

## Screenshots / Notes

This is a follow-up to the already merged proposal packet lifecycle extension PR. It is intentionally on a fresh branch from synced `main` so the PR contains only this routing-hardening delta plus the shared closeout remediation required to make closeout complete.

### CODE REVIEW

File: .octon/inputs/additive/extensions/octon-proposal-packet-lifecycle/context/routing.contract.yml [hunk 1]
```diff
@@ -144,6 +437,21 @@ dispatchers:
...
                 predicate: "equals"
                 value: "explain-proposal-packet"
+          - matcher_id: "packet-path-default-explain"
+            reason_codes: ["packet-path-read-only-default"]
+            all_of:
+              - input_name: "packet_path"
+                predicate: "present"
+              - input_name: "bundle"
+                predicate: "absent"
+              - input_name: "lifecycle_action"
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .octon/inputs/additive/extensions/octon-proposal-packet-lifecycle/context/routing.contract.yml:445]: Fixed in 22955ef. `packet-path-default-explain` now requires `source_kind` to be absent, and `source-driven-packet-creation` accepts `source_kind` even when a target `packet_path` is supplied. Added route-resolution coverage for `source_kind` + `packet_path`; local result: Passed: 179 Failed: 0.

File: .octon/inputs/additive/extensions/octon-proposal-packet-lifecycle/context/routing.contract.yml [hunk 2]
```diff
@@ -403,6 +711,8 @@ dispatchers:
                 predicate: "absent"
               - input_name: "child_packet_paths"
                 predicate: "absent"
+              - input_name: "finding_id"
+                predicate: "absent"
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .octon/inputs/additive/extensions/octon-proposal-packet-lifecycle/context/routing.contract.yml]: Fixed in 22955ef. Removed the `finding_id` absent guard from `missing-routeable-inputs`, so finding-only inputs now escalate with the actionable `missing-routeable-inputs` reason instead of falling through to `no-matching-route`. Added route-resolution coverage for this case; local result: Passed: 179 Failed: 0.

**Comment by jamesryancooper** [comment]
@codex review

File: .github/workflows/pr-auto-merge.yml [hunk 3]
```diff
@@ -248,6 +248,23 @@ jobs:
...
+            head_ref="$(jq -r '.head.ref // ""' <<<"${payload}")"
+            head_sha="$(jq -r '.head.sha // ""' <<<"${payload}")"
+
+            if [[ -z "${head_ref}" || "${head_ref}" == "null" || -z "${head_sha}" || "${head_sha}" == "null" ]]; then
+              echo "Unable to resolve PR head ref/sha for PR #${PR_NUMBER}."
+              exit 1
+            fi
+
+            git fetch --no-tags --depth=1 origin "+refs/heads/${head_ref}:refs/remotes/origin/${head_ref}"
+            git checkout --detach "${head_sha}"
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .github/workflows/pr-auto-merge.yml]: Fixed in bf0e605. `checkout_pr_head` now fetches the immutable PR head SHA directly with `git fetch --no-tags --depth=1 origin "${head_sha}"` before detached checkout, so concurrent branch movement cannot make the earlier payload SHA unavailable. Added validator/test coverage for the exact-SHA fetch. Local validation: `test-protected-ci-live-wrapper.sh` passed; `validate-git-github-workflow-alignment.sh` errors=0 warnings=0; `test-git-github-workflow-alignment.sh` Passed: 4 Failed: 0; `validate-execution-governance.sh` errors=0; `validate-phase6-simplification-deletion.sh` errors=0; `git diff --check` passed.

**Comment by jamesryancooper** [comment]
Maintainer override for solo-dev closeout: resolving Codex review threads that have been directly remediated by commits 22955ef and bf0e605, replied to with validation evidence, and revalidated with all PR checks green. This is conversation cleanup only and does not change required-check or approval gates.
